drivers/platform/aducm3029/rtc.c

```diff
@@ -0,0 +1,210 @@
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdlib.h>
+#include "error.h"
+#include "rtc.h"
+#include "rtc_extra.h"
+#include "irq.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Initialize the RTC peripheral.
+ * @param device - The RTC descriptor.
+ * @param init_param - The structure that contains the RTC initialization.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t rtc_init(struct rtc_desc **device, struct rtc_init_param *init_param)
+{
+	int32_t ret;
+	struct rtc_desc *dev;
+	struct aducm_rtc_desc *adev;
+
+	dev = (struct rtc_desc *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	adev = (struct aducm_rtc_desc *)calloc(1, sizeof(*adev));
+	if (!adev)
+		goto error_dev;
+
+	adev->memory = calloc(1, ADI_RTC_MEMORY_SIZE);
+	if(!adev->memory)
+		goto error_extra;
+
+	dev->id = init_param->id;
+	dev->freq = init_param->freq;
+	dev->load = init_param->load;
+	dev->extra = adev;
+
+	ret = adi_rtc_Open(dev->id, adev->memory, ADI_RTC_MEMORY_SIZE,
+			   &adev->instance);
+	if(ret != SUCCESS)
+		goto error_mem;
+
+	switch (dev->freq) {
```

Comment:
```suggestion
	if (dev->freq < AUDCM_32768HZ || dev->freq > AUDCM_1HZ)
	    ret = adi_rtc_SetPreScale(adev->instance, dev->freq);
	    if (ret != ADI_RTC_SUCCESS )
```

Reply:
Is the `if` not redundant if I have a `switch case` with `default`? This should cover all possible values.

Reply:
I just suggested a shorter alternative for the `switch`. In the `else `of the `if `will be the what is in the `default `now.